src/main/java/touro/chess/UndoManager.java

```diff
@@ -0,0 +1,89 @@
+package touro.chess;
+
+import java.util.Stack;
+/*
+    TO DO:
+    clean code
+    fix naming?
+    private class within public?
+    testing issues
+    other scenarios? if a piece is changed? don't know chess well enough...
+
+ */
+public class UndoManager { //don't like naming
+    Board board;
+    Stack<MoveInfo> executedMoves;
+    private class MoveInfo{
+        private Move move;
+        private boolean pieceCaptured;
```

Comment:
You don't need this boolean if you have `capturedPiece` because `capturedPiece` can be null